File: CPlus/meetingRoomII.cpp
class Solution {
public:
    int minMeetingRooms(vector<vector<int>>& intervals) {
          map<int,int> hashmap;
          for (auto interval:intervals){
              hashmap[interval[0]]++;
              hashmap[interval[1]]--;
          }
          int room=0; 
          int max_rooms=0;
          for (auto value:hashmap){
              room+=value.second;
              max_rooms=max(room,max_rooms);
          }
        
          return max_rooms;
       }          
    
};

File: Python/meetingroomII.py
class Solution(object):
    def minMeetingRooms(self, intervals):
        """
        :type intervals: List[List[int]]
        :rtype: int
        """
        if not intervals:
            return 0
        
        intervals.sort(key=lambda x:x[0])
        
        pq=[]
        room=1
        heapq.heappush(pq,intervals[0][1])
        
        for i in range(1,len(intervals)):
            if pq[0]>intervals[i][0]:
                room+=1
            else:
                heapq.heappop(pq)
            
            heapq.heappush(pq,intervals[i][1])
        
        return room